File: Introduccion/ejercicio 7/Buzon.java
package introduccion;

public class Buzon {
    private int nro;
    private int nroC; 
    private Carta[][] cartas;

    public Buzon(int nro, int nroC) {
        this.nro = nro;
        this.nroC = nroC;
        this.cartas = new Carta[3][nroC]; 
    }

    public void agregarCarta(Carta c, int fila, int col) {
        if (fila >= 0 && fila < cartas.length && col >= 0 && col < nroC) {
            cartas[fila][col] = c;
        }
    }

    public int contarCartasPorDestinatario(String destinatario) {
        int contador = 0;
        for (int i = 0; i < cartas.length; i++) {
            for (int j = 0; j < nroC; j++) {
                if (cartas[i][j] != null && cartas[i][j].getDestinatario().equalsIgnoreCase(destinatario)) {
                    contador++;
                }
            }
        }
        return contador;
    }

    public boolean eliminarCarta(String codigo) {
        for (int i = 0; i < cartas.length; i++) {
            for (int j = 0; j < nroC; j++) {
                if (cartas[i][j] != null && cartas[i][j].getCodigo().equalsIgnoreCase(codigo)) {
                    cartas[i][j] = null;
                    return true;
                }
            }
        }
        return false;
    }

    public boolean verificarRemitente(String remitente) {
        for (int i = 0; i < cartas.length; i++) {
            for (int j = 0; j < nroC; j++) {
                if (cartas[i][j] != null && cartas[i][j].getRemitente().equalsIgnoreCase(remitente)) {
                    return true;
                }
            }
        }
        return false;
    }

    public void buscarPorPalabraClave(String palabra) {
        for (int i = 0; i < cartas.length; i++) {
            for (int j = 0; j < nroC; j++) {
                if (cartas[i][j] != null && cartas[i][j].getDescripcion().toLowerCase().contains(palabra.toLowerCase())) {
                    System.out.println(cartas[i][j]);
                }
            }
        }
    }
    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder("Buzón Nº" + nro + "\n");
        for (int i = 0; i < cartas.length; i++) {
            for (int j = 0; j < nroC; j++) {
                if (cartas[i][j] != null) {
                    sb.append(cartas[i][j]).append("\n");
                }
            }
        }
        return sb.toString();
    }
}

File: Introduccion/ejercicio 7/buzon.py
from carta import Carta

class Buzon:
    def __init__(self, nro, nroC):
        self.nro = nro             
        self.nroC = nroC             
        self.cartas = [[None for _ in range(nroC)] for _ in range(3)]  

    def agregar_carta(self, c, fila, col):
        if 0 <= fila < len(self.cartas) and 0 <= col < self.nroC:
            self.cartas[fila][col] = c

    def eliminar_carta(self, codigo):
        for i in range(len(self.cartas)):
            for j in range(self.nroC):
                if self.cartas[i][j] is not None and self.cartas[i][j].get_codigo() == codigo:
                    self.cartas[i][j] = None
                    return True
        return False

    def contar_cartas_por_destinatario(self, destinatario):
        contador = 0
        for fila in self.cartas:
            for carta in fila:
                if carta is not None and carta.get_destinatario().lower() == destinatario.lower():
                    contador += 1
        return contador

    def verificar_remitente(self, remitente):
        for fila in self.cartas:
            for carta in fila:
                if carta is not None and carta.get_remitente().lower() == remitente.lower():
                    return True
        return False

    def buscar_por_palabra_clave(self, palabra):
        for fila in self.cartas:
            for carta in fila:
                if carta is not None and palabra.lower() in carta.get_descripcion().lower():
                    print(carta)

    def __str__(self):
        salida = f"Buzón Nº{self.nro}\n"
        for fila in self.cartas:
            for carta in fila:
                if carta is not None:
                    salida += str(carta) + "\n"
        return salida